### PR TITLE
Automate xray tests batch 5

### PR DESCRIPTION
**Pull request by ctschacher**
This PR adds a set of new e2e test fixtures for API definitions, applications, and group membership scenarios in the `test/platform-test/e2e/fixtures/crds` directory. These YAML files are designed to cover a variety of edge cases and behaviors related to API categories, management contexts, application members, and group memberships, improving test coverage for the platform's CRD handling logic.

See: https://gravitee.atlassian.net/browse/GKO-2814

### CODE REVIEW

File: test/platform-test/e2e/tests/gko/deployment-reconciliation/management-context-conditions.test.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    // that are present in many tests but short-lived; we spin one up here
+    // specifically to compare shapes, which is cheap.
+    // Instead of applying another CR, we rely on the fact that every GKO
+    // condition emitted by the reconciler goes through the same helper —
+    // the key check above is sufficient to detect a structural drift.
+    // We cross-check by also requiring `message` to be present or an empty
+    // string (not undefined). Some reconcilers omit it on transient states,
+    // so tolerate empty string.
+    for (const c of mgmtCtxStatus.conditions ?? []) {
+      expect(typeof (c.message ?? "")).toBe("string");
```
[the remark this answers is not in the document]
Comment (ctschacher) [reply test/platform-test/e2e/tests/gko/deployment-reconciliation/management-context-conditions.test.ts:112]: fixed

File: test/platform-test/e2e/tests/gko/api-lifecycle/v2-lifecycle-extended.test.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+    await kubectl.apply(fixturePath);
+    await kubectl.waitForCondition("apidefinition", API_NAME, "Accepted");
+
+    const apiId = (await kubectl.getStatus<{ id: string }>("apidefinition", API_NAME)).id;
+    // V2 `state: STOPPED` on the CR translates to an API whose lifecycle
+    // does not expose a STARTED state — asserting the name is stable is
+    // enough for reconciliation, and the state field is checked structurally.
+    const api = await mapi.fetchApi(apiId);
+    expect(api.name).toBe(API_NAME);
```
[the remark this answers is not in the document]
Comment (ctschacher) [reply test/platform-test/e2e/tests/gko/api-lifecycle/v2-lifecycle-extended.test.ts:188]: fixed

**Comment by ctschacher** [comment]
[E2E test suite run](https://app.circleci.com/pipelines/github/gravitee-io/gravitee-kubernetes-operator/12702/workflows/3423325b-66ab-46d9-9e9b-eef81029ea2f/jobs/81287) was successful.

**Review by a-cordier — APPROVED** [review]
